tuxemon/entity.py

```diff
@@ -226,6 +250,24 @@ def set_position(self, pos: Sequence[float]) -> None:
         self.add_collision(pos)
         self.pos_update()
 
+    def set_current_map(self, map_slug: Optional[str]) -> None:
+        """
+        Set the entity's map in the game world.
+
+        Parameters:
+            map_slug: Map to be set.
+        """
+        self._current_map = map_slug
+
+    def set_tile_position(self, tile_pos: tuple[int, int]) -> None:
+        """
+        Set the entity's tile position in the game world.
+
+        Parameters:
+            tile_pos: Tile position to be set.
+        """
+        self._tile_pos = tile_pos
```

Comment:
Should these not be decorated with the corresponding `@attribute_name.setter` decorators to match the `tile_pos` and `current_map` properties? In that case the later assignments to attributes would invoke the correct setters automatically.